src/lib/kdf/sp800_56a/sp800_56c_one_step.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+* KDF defined in NIST SP 800-56a revision 2 (Single-step key-derivation function)
+* or in NIST SP 800-56C revision 2 (Section 4 - One-Step KDM)
+*
+* (C) 2017 Ribose Inc. Written by Krzysztof Kwiatkowski.
+* (C) 2024 Fabian Albert - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/internal/sp800_56c_one_step.h>
+
+#include <botan/exceptn.h>
+#include <botan/internal/bit_ops.h>
+#include <botan/internal/fmt.h>
+#include <botan/internal/kmac.h>
+
+#include <functional>
+
+namespace Botan {
+
+namespace {
+/**
+ * @brief One-Step Key Derivation as defined in SP800-56Cr2 Section 4
+ */
+void kdm_internal(std::span<uint8_t> output_buffer,
+                  std::span<const uint8_t> z,
+                  std::span<const uint8_t> fixed_info,
+                  Buffered_Computation& h,
+                  const std::function<void(Buffered_Computation*)>& reset_h_callback) {
```

Comment:
Seems better to template this over a `T` to avoid a `dynamic_cast` for each iteration of the loop.

Also `Buffered_Computation` as a base class is anyway deprecated so we'd have to rewrite this when it's removed.

Reply:
I didn't know Buffered_Computation is deprecated. I'll use a template.